cccl

```diff
@@ -51,6 +51,48 @@ muffle=
 verbose=
 shared_index=-1
 
+# "dumpmachine" is a CC command-line argument to show the compiler's host triplet
+# "cccl" will follow "clang"'s MSVC "dumpmachine" output
+# "dumpmachinefull" is added as well to show the full MSVC version
+# Process the "dumpmachine"/"dumpmachinefull" arguments first, if they exist, and exit gracefully
+for arg in $@; do
+    if [[ $arg =~ ^-dumpmachine(full)?$ ]]; then
+        cl_output_parts=($($prog 2>&1))
+
+        msvc="msvc"
+        if [ "$arg" == "-dumpmachinefull" ]; then
+            msvc="$msvc${cl_output_parts[6]}"
+        fi
+
+        arch="$(echo ${cl_output_parts[8]} | awk '{print tolower($0)}')"
+        output="pc-windows-$msvc"
+        case $arch in
+
+        x64)
+            output="x86_64-$output"
+            ;;
+
+        # "file" command, with "libmagic", shows "Intel 80386" for "X86" architecture
+        x86)
+            output="i386-$output"
```

Comment:
Following up from #14 discussion if this should be i686 instead of i386. I'd argue for i686 using gcc as the reference, which uses i686 for "Intel 80386" output from the "file" command:

```bash
$ g++ runme.cpp -c -o runme.o && file runme.o
runme.o: ELF 64-bit LSB relocatable, x86-64, version 1 (SYSV), not stripped
$ g++ -dumpmachine
x86_64-linux-gnu

$ g++ runme.cpp -c -o runme.o && file runme.o
runme.o: ELF 32-bit LSB relocatable, Intel 80386, version 1 (SYSV), not stripped
$ gcc -dumpmachine
i686-linux-gnu
```